File: point.py
import math

class Point:
   def __init__(self, x, y):
      self.x = x
      self.y = y

   def adjacent(self, other):
      return ((self.x == other.x and abs(self.y - other.y) == 1) or
         (self.y == other.y and abs(self.x - other.x) == 1))

   def distance_sq(self, other):
      return (self.x - other.x)**2 + (self.y - other.y)**2

File: Java/Point.java
import java.lang.Math;

public class Point
{
   private int x;
   private int y;

   public Point(int x, int y)
   {
      this.x = x;
      this.y = y;
   }
  
   public int getX()
   {
      return this.x;
   }
  
   public int getY()
   {
      return this.y;
   }
  
   public boolean adjacent(Point p)
   {
      return ((this.x == p.getX() && Math.abs(this.y - p.getY()) == 1) || 
         (this.y == p.getY() && Math.abs(this.x - p.getX()) == 1));
   }

   public double distance_sq(Point p)
   {
      double xDist = this.x - p.getX();
      double yDist = this.y - p.getX();
      return (xDist * xDist) + (yDist * yDist);
   }
}
